### PR TITLE
refactor(templates): refactor Azure authentication and replace Azure tasks

### PR DESCRIPTION
**Pull request by andylangridge**
Azure authentication was previously done via Service Connections in Azure tasks. These tasks have
been replaced by scripts that make use of the Azure CLI. This means that it was possible to remove
all use of service connections within the templates (which couldn't be dynamic).

BREAKING CHANGE: Parameters changed in main wrapper templates.

https://pins-ds.atlassian.net/browse/DBO-38

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,7 +30,20 @@ resources:
 
 To use the templates in your pipeline, see the [Microsoft documentation](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops) on templates.
 
-## Versioning & Setup
+Some of the templates in this repository assume that the Azure DevOps Pipelines project is set up a certain way to make use of some of the variables. The required variables are described below, and should be added as Pipeline [Variable Groups](https://docs.microsoft.com/en-us/azure/devops/pipelines/library/variable-groups?view=azure-devops&tabs=yaml).
+
+### Azure Authentication
+
+The Azure authentication assumes that a [Service Principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals) has been set up for the project with all the required permissions.
```
Comment (mayya-bond-kc) [inline README.md:37]: It might be useful to say what exactly the required permissions might be?

File: README.md [hunk 2]
```diff
@@ -30,7 +30,20 @@ resources:
 
 To use the templates in your pipeline, see the [Microsoft documentation](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops) on templates.
 
-## Versioning & Setup
+Some of the templates in this repository assume that the Azure DevOps Pipelines project is set up a certain way to make use of some of the variables. The required variables are described below, and should be added as Pipeline [Variable Groups](https://docs.microsoft.com/en-us/azure/devops/pipelines/library/variable-groups?view=azure-devops&tabs=yaml).
```
Comment (mayya-bond-kc) [inline README.md:33]: not super clear here exactly which variables are being made use of

File: README.md [hunk 3]
```diff
@@ -30,7 +30,20 @@ resources:
...
+### Azure Authentication
+
+The Azure authentication assumes that a [Service Principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals) has been set up for the project with all the required permissions.
+
+Once the Service Principal has been set up, the following variables should be added to a variable group with the name `pipeline_secrets`:
+- `AZURE_SERVICE_PRINCIPAL_ID` - Set to the Client ID of the Service Principal
+- `AZURE_SERVICE_PRINCIPAL_SECRET` - Set to Client Secret of the Service Principal
+- `AZURE_TENANT_ID` - The ID of the Azure Active Directory tenant that contains the Service Principal
+
+In addition to the above variable, any other secrets and variables can be added to this variables group and will automatically be made available when using some of the wrapper templates in the [jobs](jobs) or [stages](stages) folders.
```
Comment (mayya-bond-kc) [inline README.md:44]: "variables" not "variable"

Comment (mayya-bond-kc) [inline README.md:44]: what do you mean by "some of the wrapper templates"? which ones?

**Review by mayya-bond-kc — APPROVED** [review]
Code looks fine to be. I just have some questions regarding the README, but they are more just suggestions

**Review by radum — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (andylangridge) [reply README.md]: Added some more detail!

File: README.md (hunk 1 above)
Comment (andylangridge) [reply README.md]: Done. Although it kind of depends on the context of the project, so hopefully I've made that a bit clearer?

**Review by mayya-bond-kc — APPROVED** [review]
niceeeeeeee

**Pull request merged**